Userland/Libraries/LibELF/DynamicLoader.cpp

```diff
@@ -240,6 +240,25 @@ void DynamicLoader::do_main_relocations()
                 m_direct_ifunc_relocations.append(relocation);
                 return;
             }
+            if (relocation.type() == R_X86_64_TLSDESC || relocation.type() == R_AARCH64_TLSDESC) {
+                // GNU ld for some reason puts TLSDESC relocations into .rela.plt
+                // https://sourceware.org/bugzilla/show_bug.cgi?id=28387
+
+                switch (do_direct_relocation(relocation, cached_result, ShouldInitializeWeak::No, ShouldCallIfuncResolver::No)) {
+                case RelocationResult::Failed:
+                    dbgln("Loader.so: {} unresolved symbol '{}'", m_filepath, relocation.symbol().name());
+                    VERIFY_NOT_REACHED();
+                case RelocationResult::ResolveLater:
+                    m_unresolved_relocations.append(relocation);
+                    break;
+                case RelocationResult::CallIfuncResolver:
+                    m_direct_ifunc_relocations.append(relocation);
+                    break;
+                case RelocationResult::Success:
+                    break;
```

Comment:
Could we `VERIFY` here that the result of this relocation is a success? We should never need to look up a possibly-undefined symbol to handle TLSDESC.